api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValue.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.spi;
+
+import java.util.Objects;
+
+/**
+ * The ConfigValue is a metadata object that holds additional information after the lookup of a configuration.
+ * <p>
+ *
+ * Right now, it is able to hold information like the configuration name, value, the Config Source from where
+ * the configuration was loaded, the ordinal of the Config Source and a line number from where the configuration was
+ * read if exists.
+ * <p>
+ *
+ * This is used together with {@link ConfigSource} to expose the Configuration lookup metadata.
+ *
+ * @author <a href="mailto:radcortez@yahoo.com">Roberto Cortez</a>
+ */
+public class ConfigValue {
```

Comment:
We can do as many changes as we like.

The main reason why `ConfigValue` is a class is so users can do what they like with it and implement it in `ConfigSource`. Similar to `Response` in JAX-RS.

There is some discussion about adding this or not to `ConfigSource` so we probably need to resolve that first, because it will help clear out the outcome of this.